File: 250305/G or H 3/G-or-H-3.cpp
#include <iostream>
#include <cmath>

using namespace std;

int N, K;
int arr[10001] = {0, };
int x;
char c;
int answer = 0;

int main() {
    cin >> N >> K;

    for (int i = 0; i < N; i++) {
        cin >> x >> c;
        if (c == 'G') arr[x-1] = 1;
        else arr[x-1] = 2;
    }

    for (int i = 0; i < K; i++) {
        int count = 0;
        for (int j = i; j <= N + i; j++) {
            count += arr[j];
        }
        answer = max(answer, count);
    }

    cout << answer;

    return 0;
}

File: 250309/G or H 3/G-or-H-3.py
n, k = map(int, input().split())
cnt_array = [0] * 10010
for _ in range(n):
    pos, char = input().split()
    if char == 'G':
        cnt_array[int(pos)] = 1
    else:
        cnt_array[int(pos)] = 2

# 완전탐색 진행 n - k + 1번 반복 -> O(n -k + 1)
# 사진크기만큼의 k번 반복 -> O(k)
# O(nk)
# 만약 counting array 생성 시 O(n) 수행 -> 값을 받을 때 바로 만들기
# O(nk + n) 연산 1억회 안 넘어서 가능함

answer = 0
for i in range(0, 10010 - k):
    count = 0
    for j in range(i, i + k + 1):
        if cnt_array[j]:
            count += cnt_array[j]
    answer = max(answer, count)

print(answer)